### ISSUE TITLE
Lockfile location/filename should be customizable

### ISSUE THREAD

**Issue by gabarmas** [gradle/gradle#5639]
### Expected Behavior
I'd like the location or the filename (or both) of the lockfiles to be customizable.
Something like:
```groovy
dependencyLocking {
    lockAllConfigurations()
    lockFileLocation = "gradle/dependency-locks/${scalaVersion}/"
}
//Or
configurations {
    compileClasspath {
        resolutionStrategy.activateDependencyLocking( "gradle/dependency-locks/${scalaVersion}/compile.lockfile")
    }
}
```
### Current Behavior
Right now, the lockfiles are always saved into `gradle/dependency-locks/<configuration>.lockfile`.

### Context
This feature is specially useful for scala projects. In scala, is best practice to name artifacts with the scala version they were compiled with, see for example [akka].(https://mvnrepository.com/artifact/com.typesafe.akka/akka-actor)
```
    compile "com.typesafe.akka:akka-actor${scalaVersion}:2.5.+"
```

It's also common place to have some sort of crosscompilation in scala projects, either by [custom plugins](https://github.com/ADTRAN/gradle-scala-multiversion-plugin) or by having multiple builds in CI, as simple as `./gradlew build -PscalaVersion=2.11`.

So the dependencies change depending on the scala version. 
Right now, that block us from using the dependency lock feature, as we would be overriding or consuming the wrong libraries (or the correct libraries with the wrong scala version)

### Your Environment
Gradle 4.8

**Comment by melix** [comment]
It's a reasonable thing to be able to use a different lock file based on some runtime environment. Scala here is just an example (that could be solved differently, it's just how you model the build).

That's basically why we have a `dependencyLocking` extension: it's meant to add more configuration options as we have more requests. That said we had in mind to be able to provide a different locking provider, not necessarily a file.

**Comment by prageeth** [comment]
@melix I have a similar issue here, and I would like to configure the `dependencyLocking` extension to allow a different "lock state" depending on the environment. Is there an example I can look at? Thanks!

**Comment by ljacomet** [comment]
What @melix hinted at is a possible _future_ solution, not something you can do right now.

**Comment by ljacomet** [comment]
This will be resolved as part of #11881 and is planned to be released with Gradle 6.4.
See #12561 that makes it possible.

**Comment by ljacomet** [comment]
The feature is now merged to `master`
